lib/propolis/src/accessors.rs

```diff
@@ -261,18 +286,26 @@ impl<T> Tree<T> {
         self.root_key() == node.into()
     }
 
-    fn poison(&mut self) -> Option<Arc<T>> {
-        // Remove the resource from the tree...
-        let resource = self.resource_root.take();
+    fn set_root_resource(
+        &mut self,
+        new_root: Option<T::Root>,
+    ) -> Option<T::Root> {
+        // Swap out the existing root resource
+        let old = std::mem::replace(&mut self.res_root, new_root);
 
         // ... and poison all nodes too
```

Comment:
more of a cache invalidation than a poison, now?

Reply:
Done.